lib/main_common.pm

```diff
@@ -186,6 +186,11 @@ sub setup_env {
     }
 }
 
+sub data_integrity_is_applicable {
+  # For other backends like Xen, zKVM or Hyper-V we need check it after the image is downloaded
+  return (grep { /^CHECKSUM_/ } keys %bmwqemu::vars) && check_var('BACKEND', 'qemu');
```

Comment:
nitpicking, but let's change order of conditions, as search of variables is more expensive, and with lazy evaluation we would skip it in case of non-qemu backend..

Reply:
+1. You can also put the `check_var` on separate line under the comment as a `return check_var`, so it's obvious that the comment is intended only for the second part of the condition.

Reply:
right, in this case one-liner doesn't help with the comment